src/libraries/System.Globalization/tests/CultureInfo/GetCultureInfo.cs

```diff
@@ -139,5 +140,60 @@ public void PredefinedCulturesOnlyEnvVarTest(string predefinedCulturesOnlyEnvVar
                 }
             }, cultureName, predefinedCulturesOnlyEnvVar, new RemoteInvokeOptions { StartInfo = psi }).Dispose();
         }
+
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        public void TestAllowInvariantCultureOnly(bool enableInvariant, bool enableInvariantOnly)
+        {
+            var psi = new ProcessStartInfo();
+            psi.Environment.Clear();
+
+            if (enableInvariant)     { psi.Environment.Add("DOTNET_SYSTEM_GLOBALIZATION_INVARIANT",              "true"); }
+            if (enableInvariantOnly) { psi.Environment.Add("DOTNET_SYSTEM_GLOBALIZATION_INVARIANT_CULTURE_ONLY", "true"); }
+
+            RemoteExecutor.Invoke((invarintEnabled, invariantOnlyEnabled) =>
+            {
+                bool restrictedMode = bool.Parse(invarintEnabled) && bool.Parse(invariantOnlyEnabled);
+
+                // First ensure we can create the current cultures regaldless of the mode we are in
+                Assert.NotNull(CultureInfo.CurrentCulture);
+                Assert.NotNull(CultureInfo.CurrentUICulture);
+
+                // Invariant culture should be valid all the time
```

Comment:
Should we test a couple basic operations work with invariant mode here? string operations, calendar, etc? Just super basic things

Reply:
I added a couple of more cases to test invariant.